### ISSUE TITLE
Gradients do not exist for these variables

### ISSUE THREAD

**Issue by adithyak-hub** [tensorlayer/srgan#210]
WARNING:tensorflow:Gradients do not exist for variables ['batchnorm2d_1/moving_mean:0', 'batchnorm2d_1/moving_var:0', 'batchnorm2d_2/moving_mean:0', 'batchnorm2d_2/moving_var:0', 'batchnorm2d_3/moving_mean:0', 'batchnorm2d_3/moving_var:0', 'batchnorm2d_4/moving_mean:0', 'batchnorm2d_4/moving_var:0', 'batchnorm2d_5/moving_mean:0', 'batchnorm2d_5/moving_var:0', 'batchnorm2d_6/moving_mean:0', 'batchnorm2d_6/moving_var:0', 'batchnorm2d_7/moving_mean:0', 'batchnorm2d_7/moving_var:0', 'batchnorm2d_8/moving_mean:0', 'batchnorm2d_8/moving_var:0', 'batchnorm2d_9/moving_mean:0', 'batchnorm2d_9/moving_var:0', 'batchnorm2d_10/moving_mean:0', 'batchnorm2d_10/moving_var:0', 'batchnorm2d_11/moving_mean:0', 'batchnorm2d_11/moving_var:0', 'batchnorm2d_12/moving_mean:0', 'batchnorm2d_12/moving_var:0', 'batchnorm2d_13/moving_mean:0', 'batchnorm2d_13/moving_var:0', 'batchnorm2d_14/moving_mean:0', 'batchnorm2d_14/moving_var:0', 'batchnorm2d_15/moving_mean:0', 'batchnorm2d_15/moving_var:0', 'batchnorm2d_16/moving_mean:0', 'batchnorm2d_16/moving_var:0', 'batchnorm2d_17/moving_mean:0', 'batchnorm2d_17/moving_var:0', 'batchnorm2d_18/moving_mean:0', 'batchnorm2d_18/moving_var:0', 'batchnorm2d_19/moving_mean:0', 'batchnorm2d_19/moving_var:0', 'batchnorm2d_20/moving_mean:0', 'batchnorm2d_20/moving_var:0', 'batchnorm2d_21/moving_mean:0', 'batchnorm2d_21/moving_var:0', 'batchnorm2d_22/moving_mean:0', 'batchnorm2d_22/moving_var:0', 'batchnorm2d_23/moving_mean:0', 'batchnorm2d_23/moving_var:0', 'batchnorm2d_24/moving_mean:0', 'batchnorm2d_24/moving_var:0', 'batchnorm2d_25/moving_mean:0', 'batchnorm2d_25/moving_var:0', 'batchnorm2d_26/moving_mean:0', 'batchnorm2d_26/moving_var:0', 'batchnorm2d_27/moving_mean:0', 'batchnorm2d_27/moving_var:0', 'batchnorm2d_28/moving_mean:0', 'batchnorm2d_28/moving_var:0', 'batchnorm2d_29/moving_mean:0', 'batchnorm2d_29/moving_var:0', 'batchnorm2d_30/moving_mean:0', 'batchnorm2d_30/moving_var:0', 'batchnorm2d_31/moving_mean:0', 'batchnorm2d_31/moving_var:0', 'batchnorm2d_32/moving_mean:0', 'batchnorm2d_32/moving_var:0', 'batchnorm2d_33/moving_mean:0', 'batchnorm2d_33/moving_var:0'] when minimizing the loss.
This is shown after every step of an epoch
What to do?

**Comment by zsdonghao** [comment]
use   `model.trainable_weights`

**Comment by adithyak-hub** [comment]
Instead of G.trainable_weights?
Because I change it to G.weights from G.trainable_weights because it showed trainable_weights cannot be loaded

**Comment by adithyak-hub** [comment]
> use `model.trainable_weights`

May I know where to use this?

**Comment by dadajuan** [comment]
> Instead of G.trainable_weights?
> Because I change it to G.weights from G.trainable_weights because it showed trainable_weights cannot be loaded

i meet the same wrong, do you fix it?
